tests/mammotion-cli.md

```diff
@@ -0,0 +1,415 @@
+
+**Usage Examples:**
+```
+MowerControl> rawcmd get_report_cfg
+MowerControl> rawcmd set_led_shine '{"shine_mode":1, "custom_time":{"start_hour":20, "start_min":0, "end_hour":6, "end_min":0}}'
+MowerControl> rawcmd allpowerfull_rw rw_id=5 rw=1 context=1
+```
+> **Note:** The exact parameters and their structure depend on the specific command. Refer to `PyMammotion` documentation or source code. Type `rawcmd` alone for examples.
+
+---
+
+### `help`
+Displays a summary of available commands and their syntax.
+
+**Usage:**
+```
+MowerControl> help
+```
+
+---
+
+### `exit`
+Disconnects from the mower, stops continuous reporting and background tasks, and closes the application gracefully.
+
+**Usage:**
+```
+MowerControl> exit
+```
+
+---
+
+## 7. Understanding Status and Output
+
+*   **Log Messages:** The script logs its actions, status updates, and any errors to the console. The default level is `INFO`. Set `LOG_LEVEL="DEBUG"` (see [Setup](#3-setup)) for more verbose output. The script attempts to handle known benign errors (like certain `JSONDecodeError`s from background MQTT tasks) gracefully.
+*   **Periodic Status:** The `STATUS:` line that appears periodically shows a snapshot of the mower's state, including activity, battery, RTK, progress, and current area. This is driven by the `periodic_status_monitor`.
+*   **Notifications:** The script includes simple logic to log a "NOTIFICATION" if a task that was previously working at high progress changes to an idle/completed state, suggesting task completion.
+
+## 8. Troubleshooting
+
+*   **Connection Issues:**
+    *   Double-check email/password (case-sensitive).
+    *   Ensure your Luba mower is online and has a stable internet connection (WiFi).
+    *   Check for firewall issues on your computer or network.
+    *   Use the `connect` command to re-attempt connection or restart the script.
+*   **Commands Not Working:**
+    *   Ensure the mower is in an appropriate state for the command.
+    *   Allow a few seconds for commands to be processed and status to update.
+    *   Use `raw_status` to inspect the mower's complete state.
+    *   Use `status` to check the mower's current activity.
+*   **Incomplete Plan List (`maps` command):**
+    *   Run `sync_maps`. Wait 10-15 seconds. Then run `maps` again.
+*   **`AttributeError` or Command Not Found:**
+    This usually means the command name used does not map to a known function in the `PyMammotion` library as expected by the script. Check for typos.
+
+## 9. Advanced Notes
+
+### Device-Specific Behavior
+
+*   **`mow_areas` Command:**
+    *   **Yuka:** If a Yuka device is detected, the `blade_height` for the custom job will automatically be set to `-10` as required.
+    *   **Luba 1:** Specific adjustments may be made to `GenerateRouteInformation` parameters (e.g., `toward_mode`, `toward_included_angle` set to 0). After starting a job, an additional `set_blade_control(on_off=1)` command is sent to help ensure blades engage.
```

Comment:
Don't call `set_blade_control` it tends to stuff up the mower and usually requires a restart. If blades didn't start something else is wrong with the GenerateRouteInformation values